src/entities/SecurityToken/Controller.ts

```diff
@@ -41,4 +41,28 @@ export class Transfers extends SubModule {
 
     return procedure.prepare();
   };
+
+  /**
+   * Redeem (burn) an amount of tokens from a token holder. Only the Security Token's controller can call this
+   * This operation is subject to transfer restrictions and the amount is limited by the token holder's balance.
+   * `balanceOf(tokenHolder)` tokens) and potentially also need to respect other transfer restrictions.
+   *
+   * @param amount of tokens to be redeemed
+   * @param address of the token holder
+   * @param reason optional message to describe why the redemption occurred
+   * @param data optional data used to validate the transfer
+   */
+  public controllerRedeem = async (args: {
+    amount: BigNumber;
+    from: string;
+    reason?: string;
+    data?: string;
+  }) => {
+    const { amount, from, reason = '', data = '' } = args;
+    const { symbol } = this.securityToken;
+
+    const procedure = new ControllerRedeem({ symbol, amount, from, reason, data }, this.context);
```

Comment:
Since reason and data are already optional in the procedure and they have default assignments over there, here you can just do `new ControllerRedeem({ symbol, ...args }, this.context);` and skip the destructuring in the first line